src/protocols/secure_channel/tests/TestCheckinMsg.cpp

```diff
@@ -40,106 +39,228 @@ using namespace chip::Protocols;
 
-        Hmac128KeyHandle hmac128KeyHandle;
-        NL_TEST_ASSERT_SUCCESS(inSuite, keystore.CreateKey(hmacKeyMaterial, hmac128KeyHandle));
+    // Create output buffer with 0 size
+    MutableByteSpan output;
 
-        // As of now passing an empty key handle while using PSA crypto will result in a failure.
-        // However when using OpenSSL this same test result in a success.
-        // Issue #28986
+    NL_TEST_ASSERT(inSuite, CHIP_ERROR_BUFFER_TOO_SMALL == GenerateAndVerifyPayload(inSuite, output, vector));
+}
 
-        // Aes128KeyHandle emptyKeyHandle;
-        // err = CheckinMessage::GenerateCheckinMessagePayload(emptyKeyHandle, counter, userData, outputBuffer);
-        // ChipLogError(Inet, "%s", err.AsString());
-        // NL_TEST_ASSERT(inSuite, (CHIP_NO_ERROR == err));
+/**
+ * @brief Test verifies that the Check-In Message generations returns an error if the AesKeyHandle is empty
+ */
+void TestCheckInMsg::TestCheckInMessageGenerate_EmptyAesKeyHandle(nlTestSuite * inSuite, void * inContexT)
+{
+    TestSessionKeystoreImpl keystore;
+    CheckIn_Message_test_vector vector = checkIn_message_test_vectors[0];
 
-        // Testing empty application data
-        ByteSpan emptyData;
-        err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, emptyData, outputBuffer);
-        NL_TEST_ASSERT(inSuite, (CHIP_NO_ERROR == err));
+    // Create output buffer
+    uint8_t buffer[300] = { 0 };
+    MutableByteSpan output(buffer, sizeof(buffer));
 
-        // Testing empty output buffer
-        MutableByteSpan empty;
-        err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, emptyData, empty);
-        NL_TEST_ASSERT(inSuite, (CHIP_ERROR_BUFFER_TOO_SMALL == err));
+    // Force output buffer to the payload size
+    output.reduce_size(vector.payload_len);
 
-        // Test output buffer smaller than the ApplicationData
-        userData = chip::ByteSpan(veryLargeBuffer, sizeof(veryLargeBuffer));
-        err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, userData, outputBuffer);
-        NL_TEST_ASSERT(inSuite, (CHIP_ERROR_BUFFER_TOO_SMALL == err));
+    // Empty AES Key handle
+    Aes128KeyHandle aes128KeyHandle;
 
-        // Cleanup
-        keystore.DestroyKey(aes128KeyHandle);
-        keystore.DestroyKey(hmac128KeyHandle);
-    }
+    Symmetric128BitsKeyByteArray hmacKeyMaterial;
+    memcpy(hmacKeyMaterial, vector.key, vector.key_len);
+
+    Hmac128KeyHandle hmac128KeyHandle;
+    NL_TEST_ASSERT_SUCCESS(inSuite, keystore.CreateKey(hmacKeyMaterial, hmac128KeyHandle));
+
+    // Create application data ByteSpan
+    ByteSpan applicationData(vector.application_data, vector.application_data_len);
+
+    /*
+        Passing an empty key handle while using PSA crypto will result in a failure.
+        When using OpenSSL this same test result in a success.
+        Issue #28986
```

Comment:
Suggest `TODO(#28986): xxxx` format for these things, and wrap with `#if 0`. OK to do in follow-up that touches this test.

Reply:
Will be done in the follow up PR with the Parsing tests.